### DISCUSSION TITLE
Project Structure

### DISCUSSION THREAD

**Discussion by wireboy5** [Slow-Scallion4183/Adventure-Colussus#16] [Ideas]
May want to get this working sooner rather than later, as once your project gets larger it will take more time to do.

Structuring your project is important. I recommend you read [this here](https://docs.python-guide.org/writing/structure/) for more information on it.

For this project, a good structure would look like this:

- adventure_colussus
    - \_\_init\_\_.py
    - \_\_main\_\_.py
- Adventure_Colussus_game_ideas.txt
- Contributors.md
- README.md
- .gitignore

@Slow-Scallion4183  It would be best if you did this yourself because It is a good learning experience.

\_\_init\_\_.py would contain the library functions of your code, for example, print_text

\_\_main\_\_.py would contain the actual functionality of adventue_colussus. For example, the main adventure.

You would import files in \_\init\_\_.py from \_\_main\_\_.py simply by importing from adventure_colussus:

```python
import adventure_colussus
# or
from adventure_colussus import *
```

For running the program, you would go to the main project directory (where README is) and run this (or a similar) command:
```bash
python -m adventure_colussus
```

In the future, you may also want to add a setup.py file for people installing your game, and that is detailed in the link above.

Happy Coding :)

**Comment by Slow-Scallion4183** [comment]
Thanks for that! Ill check it out. Souns like a good idea ☺
